File: ingraph-compiler/ingraph-compiler-cypher2search/src/main/java/ingraph/cypher2relalg/util/GrammarUtil.java
package ingraph.cypher2relalg.util;

import org.eclipse.emf.ecore.EObject;
import org.slizaa.neo4j.opencypher.openCypher.Create;
import org.slizaa.neo4j.opencypher.openCypher.Delete;

/**
 * Utility class to use for the slizaa openCypher Xtext grammar.
 */
@SuppressWarnings("all")
public class GrammarUtil {
  /**
   * Determines whether a clause is a CREATE/UPDATE/DELETE or not.
   * 
   * FIXME: UPDATE is not handled yet.
   */
  public static boolean isCudClause(final EObject o) {
    boolean _switchResult = false;
    boolean _matched = false;
    if (o instanceof Create) {
      _matched=true;
      _switchResult = true;
    }
    if (!_matched) {
      if (o instanceof Delete) {
        _matched=true;
        _switchResult = true;
      }
    }
    if (!_matched) {
      _switchResult = false;
    }
    return _switchResult;
  }
}

File: compiler/src/main/scala/ingraph/compiler/cypher2qplan/util/GrammarUtil.scala
package ingraph.compiler.cypher2qplan.util


import org.eclipse.emf.ecore.EObject
import org.slizaa.neo4j.opencypher.{openCypher => oc}

/**
 * Utility object to use for the slizaa openCypher Xtext grammar.
 */
object GrammarUtil {
	/**
	 * Determines whether a clause is a CREATE/UPDATE/DELETE or not.
	 *
	 * FIXME: UPDATE is not handled yet.
	 */
	def isCudClause(o: EObject): Boolean = {
		o match {
			case _: oc.Create => true
			case _: oc.Delete => true
			case _: oc.Merge  => true
			case _: oc.Remove => true
			case _: oc.Set    => true
			case _ => false
		}
	}
}
